static/app/utils/profiling/profile/importProfile.tsx

```diff
@@ -96,8 +95,106 @@ export function importProfile(input: ProfileInput, traceID: string): ProfileGrou
   }
 }
 
+function sortSamples(
+  a: {stack: number[]; weight: number},
+  b: {stack: number[]; weight: number}
+) {
+  const max = Math.max(a.stack.length, b.stack.length);
+
+  for (let i = 0; i < max; i++) {
+    if (a.stack[i] === undefined) {
+      return -1;
+    }
+    if (b.stack[i] === undefined) {
+      return 1;
+    }
+    if (a.stack[i] === b.stack[i]) {
+      continue;
+    }
+    return a.stack[i] - b.stack[i];
+  }
+  return 0;
+}
+
+function sortJSSelfProfileSamples(samples: JSSelfProfiling.Trace['samples']) {
+  return [...samples].sort((a, b) => {
+    return a.stackId - b.stackId;
+  });
+}
+
+function sortSentrySampledProfileSamples(
+  samples: Profiling.SentrySampledProfile['profile']['samples']
+) {
+  return [...samples].sort((a, b) => {
+    return a.stack_id - b.stack_id;
+  });
+}
+
+function sortSchemaSamples(
+  profile: Readonly<Profiling.Schema['profiles'][0]>
+): {stack: number[]; weight: number}[] | null {
+  if (isSampledProfile(profile)) {
+    const weightsWithSamples = profile.samples.map((stack, index) => {
+      return {
+        stack,
+        weight: profile.weights[index],
+      };
+    });
+
+    return weightsWithSamples.sort(sortSamples);
+  }
+  if (isEventedProfile(profile)) {
+    return null;
+  }
+
+  throw new TypeError('Unknown profile type');
+}
+
+// Presort samples so that they are naturally collapsed into a graph.
+// it is important that we do not mutate the original input.
+export function sortProfileSamples<T extends Profiling.ProfileInput>(
+  input: Readonly<T>
+): T {
+  if (isSchema(input)) {
+    const output: T = {...input, profiles: []};
+
+    for (let i = 0; i < input.profiles.length; i++) {
+      // @ts-expect-error we are assigning to a copy
```

Comment:
What's the error here?

Reply:
T is inferred as readonly so when we assign to it it screams, but we are creating a copy here anyways